src/models/PenaltyIdentifierSchemaFactory.ts

```diff
@@ -11,46 +8,21 @@ export class PenaltyIdentifierSchemaFactory {
 
     constructor(companyNumberPrefixes: string) {
 
-        if (companyNumberPrefixes.length === 0) {
-            throw new Error('Company prefixes must not be empty. e.g. NI,SC,SO');
-        }
-
-        if (companyNumberPrefixes === '*') {
-            this.companyNumberRegex = /^[A-Za-z0-9]{8}$/i;
-        } else {
+        const listCheckRegex: RegExp = /^([A-Z][A-Z]?)\b(,[A-Z][A-Z]?)*$/i;
 
-            const prefixesArray: string[] = companyNumberPrefixes.split(',');
-
-            const singleCharacterPrefixArray: string[] = prefixesArray.filter(prefix => prefix.length === 1);
-            const doubleCharacterPrefixArray: string[] = prefixesArray.filter(prefix => prefix.length === 2);
+        if (!listCheckRegex.test(companyNumberPrefixes)) {
+            throw new Error('Prefix list formatting error. Make sure list is comma separated e.g. NI,SI,R');
+        }
 
-            const prefixes: PrefixMap = {
-                singleCharacter: {
-                    lowerCase: this.makeLowerCase(singleCharacterPrefixArray),
-                    upperCase: this.makeUpperCase(singleCharacterPrefixArray)
-                },
-                doubleCharacter: {
-                    lowerCase: this.makeLowerCase(doubleCharacterPrefixArray),
-                    upperCase: this.makeUpperCase(doubleCharacterPrefixArray)
-                }
-            };
+        const prefixesArray: string[] = companyNumberPrefixes.split(',');
 
-            const singleCharacterRegex = `(${[
-                prefixes.singleCharacter.lowerCase.reduceRight((p, c) => `${p}|${c}`, '').substr(1),
-                prefixes.singleCharacter.upperCase.reduceRight((p, c) => `${p}|${c}`, '')
-            ].join('')})[0-9]{1,7}`;
+        const prefixRegex = `(${prefixesArray.reduceRight((p, c) => `${p}|${c}`, '').substr(1)})[0-9]{1,7}`;
```

Comment:
does this still work with both single and double prefix characters?

Reply:
looking at the regex at the end: `[0-9]{1,7}`

Reply:
Yes you're right. Good spot